modules/core/arrow-instances/src/main/kotlin/arrow/instances/option.kt

```diff
@@ -62,6 +62,12 @@ interface OptionEqInstance<A> : Eq<Option<A>> {
 
 }
 
+@instance(Option::class)
+interface OptionHashInstance<A> : OptionEqInstance<A>, Hash<Option<A>> {
```

Comment:
i'm fixing this. i think it should be more like:

```
@instance(Option::class)
interface OptionHashInstance<A> : Eq<Option<A>>, Hash<Option<A>> {

    fun HASH(): Hash<A>

    override fun hash(a: Option<A>): Int = when (a) {
       None -> None.hashCode()
       is Some -> HASH().hash(a.t).hashCode()
    }
}
```

but then i get:

```Object is not abstract and does not implement abstract member public abstract fun eqv(a: Option<A>, b: Option<A>): Boolean defined in arrow.instances.OptionHashInstance```

any idea?

Reply:
what happens if you provide?
```kotlin
@instance(Option::class)
interface OptionHashInstance<A> : OptionEqInstance<A>, Hash<Option<A>> {

    fun HS(): Hash<A>

    override fun EQ(): Eq<A> = HS()

    override fun hash(a: Option<A>): Int = when (a) {
       None -> None.hashCode()
       is Some -> HASH().hash(a.t).hashCode()
    }
}
```

Reply:
nice, that works. thanks!